[Serve][Doc] Add Application serve examples. (#33164)
ray-project/ray: doc/source/serve/tutorials/text-classification.md

> (serve-text-classification-tutorial)=

Reviewer: Could you apply the same wording suggestions I made for `object-detection.md` to this file also?

> # Serving Distilbert Model

Reviewer 2: ```suggestion
# Serving a Distilbert Model
```

> Build a IMDB review classification application with Ray Serve.

Reviewer 2: ```suggestion
This example builds an IMDB review classification application with Ray Serve.
```

To run this example, you will need to install the following:

```bash
pip install "ray[serve]" requests torch transformers
```

> This example uses the [distilbert-base-uncased](https://huggingface.co/docs/transformers/tasks/sequence_classification) model and FastAPI to build the example. Save the following code to a file named distilbert_app.py. 

Reviewer 2: ```suggestion
This example uses the [distilbert-base-uncased](https://huggingface.co/docs/transformers/tasks/sequence_classification) model and FastAPI. Save the following code to a file named distilbert_app.py:
```

> Serve Code:

Reviewer 2: ```suggestion
Use the following Serve code:
```

```{literalinclude} ../doc_code/distilbert.py
```

Use `serve run distilbert_app:entrypoint` to start the serve application.

:::{note}

> When `min_replica = 0` in the autoscaling configuration, the Serve application spawns a replica when a request arrives. If there are no requests, it downscales replicas to 0 to save GPU resources.

Reviewer 2: ```suggestion
The autoscaling config sets `min_replicas` to 0, which means the deployment starts with no `ObjectDetection` replicas. These replicas spawn only when a request arrives. When no requests arrive after a certain period, Serve downscales `ObjectDetection` back to 0 replica to save GPU resources.
```

:::

> These messages are expected in the output.

Reviewer 2: ```suggestion
You should see the following messages in the output:
```

```text
(ServeController pid=362, ip=10.0.44.233) INFO 2023-03-08 16:44:57,579 controller 362 http_state.py:129 - Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-7396d5a9efdb59ee01b7befba448433f6c6fc734cfa5421d415da1b3' on node '7396d5a9efdb59ee01b7befba448433f6c6fc734cfa5421d415da1b3' listening on '127.0.0.1:8000'
(ServeController pid=362, ip=10.0.44.233) INFO 2023-03-08 16:44:57,588 controller 362 http_state.py:129 - Starting HTTP proxy with name 'SERVE_CONTROLLER_ACTOR:SERVE_PROXY_ACTOR-a30ea53938547e0bf88ce8672e578f0067be26a7e26d23465c46300b' on node 'a30ea53938547e0bf88ce8672e578f0067be26a7e26d23465c46300b' listening on '127.0.0.1:8000'
(HTTPProxyActor pid=439, ip=10.0.44.233) INFO:     Started server process [439]
(HTTPProxyActor pid=5779) INFO:     Started server process [5779]
(ServeController pid=362, ip=10.0.44.233) INFO 2023-03-08 16:44:59,362 controller 362 deployment_state.py:1333 - Adding 1 replica to deployment 'APIIngress'.
2023-03-08 16:45:01,316 SUCC <string>:93 -- Deployed Serve app successfully.
```

> Use the following code to send requests.

Reviewer 2: ```suggestion
Use the following code to send requests:
```

```python
import requests

prompt = "This was a masterpiece. Not completely faithful to the books, but enthralling from beginning to end. Might be my favorite of the three."
input = "%20".join(prompt.split(" "))
resp = requests.get(f"http://127.0.0.1:8000/classify?sentence={prompt}")
print(resp.status_code, resp.json())
```

> The output of the client code is response status code, label (positive in this example) and its score.

Reviewer: ```suggestion
The output of the client code is the response status code, the label (positive in this example), and the label's score.
```
